theano/sandbox/cuda/extra_ops.py

```diff
@@ -37,8 +37,11 @@ def make_node(self, x):
         if x.ndim > GpuCumsum.SUPPORTED_NDIMS:
             raise NotImplementedError('Only cumsum on 1D, 2D and 3D array are supported right now!')
 
-        if self.axis >= x.ndim:
+        if self.axis >= x.ndim or self.axis < -x.ndim:
             raise ValueError('axis(={1}) out of bounds'.format(self.axis))
+        elif self.axis < 0:
+            # Convert negative axis to positive axis.
+            self.axis += x.ndim
```

Comment:
This is wrong.  'axis' is global for all places that use this op instance, not just for the node you are building right now.